compiler/src/dotty/tools/dotc/sbt/ExtractAPI.scala

```diff
@@ -343,6 +344,12 @@ private class ExtractAPICollector(implicit val ctx: Context) extends ThunkHolder
     }
   }
 
+  // Hack to represent dotty types which don't have an equivalent in xsbti
+  def combineApiTypes(apiTps: api.Type*): api.Type = {
```

Comment:
I have no idea what this is supposed to achieve.

Reply:
Basically fitting a square peg in a round hole: sbt has its own representation of types which doesn't include anything like TypeArgRef, but we can still get incremental compilation to work correctly if we can find a bijection between dotty's types and sbt's types, since the only thing that matter is that the hash of the sbt type changes when the dotty type changes. Don't worry too much about it :).